src/decisionengine/framework/dataspace/dataspace.py

```diff
@@ -51,6 +51,10 @@ def create_datasource(module_name, class_name, config):
             py_module = importlib.import_module(module_name)
             clazz = getattr(py_module, class_name)
             ds = clazz(config)
+        elif 'SQLAlchemy' in class_name:
+            # There is a good chance we are forked off to another process
+            # and already hace cached connections, so reset the connections.
+            ds.reset_connections()
```

Comment:
I wonder if the check can be more general (e.g.):

```python
if not ds:
  ...
  ds = clazz(config)
elif config.get('reset_connections', True):
  ds.reset_connections()
```

This prevents the creep of datasource implementations into the base class.  Granted, here we're only making a string comparison and not introducing an actual dependency on SQLAlchemy.  It's still preferable, however, to keep the base free of implementation concepts if we can.

Reply:
I wonder if we should add `reset_connections` to the ABC and put a stub in the existing postgres class.  Similar to how it currently does nothing with `connect()`.... thoughts?

Reply:
I could go with that.